src/tests/multihost/ad/test_login_attr.py

```diff
@@ -0,0 +1,336 @@
+:testtype: functional
+"""
+
+
+import pytest
+from pexpect import pxssh
+from sssd.testlib.common.utils import sssdTools
+
+
+@pytest.fixture(scope="function", name="sssd_domain")
+def fixture_sssd_domain(session_multihost, request):
+    """Change domain to sssd_domain"""
+    session_multihost.client[0].run_command(
+        'cp -f /etc/sssd/sssd.conf /etc/sssd/sssd.conf.sssd_domain',
+        raiseonerr=False
+    )
+    client = sssdTools(session_multihost.client[0], session_multihost.ad[0])
+    dom_section = f'domain/{client.get_domain_section_name()}'
+    session_multihost.client[0].run_command(
+        f'sed -i  "s|{dom_section}|domain/sssd_domain|g" /etc/sssd/sssd.conf',
+        raiseonerr=False
+    )
+    client.sssd_conf('sssd', {'domains': 'sssd_domain', })
+
+    def remove_sssd_domain():
+        """ Remove sssd_domain"""
+        session_multihost.client[0].run_command(
+            'mv -f /etc/sssd/sssd.conf.sssd_domain /etc/sssd/sssd.conf',
+            raiseonerr=False
+        )
+    request.addfinalizer(remove_sssd_domain)
+
+
+@pytest.mark.tier1_3
+@pytest.mark.adloginattr
+@pytest.mark.usefixtures("joinad")
+class TestADLoginAttributes:
+    """Automated Test Cases for AD Schema ported from bash"""
+
+    @staticmethod
+    def test_0001_login_by_samaccountname(multihost, create_aduser_group):
+        """test_0001_login_by_samaccountname
+
+        :title: IDM-SSSD-TC: ad_provider: ad_login_attr: Default behaviour
+          by login with sAMAccountName
+        :id: ba681368-eefe-459f-906e-1b32d405d496
+        :setup:
+          1. Configure sssd with ldap_user_principal = userPrincipalName
+        :steps:
+          1. Run getent passwd for the user.
+          2. Run su for the user with short name (without domain).
+          3. Run ssh for the user with short name (without domain).
+        :expectedresults:
+          1. User is found.
+          2. Su command succeeds.
+          3. Ssh command succeeds.
+        :customerscenario: False
+        """
+        # Configure sssd
+        multihost.client[0].service_sssd('stop')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        client.backup_sssd_conf()
+        dom_section = f'domain/{client.get_domain_section_name()}'
+        sssd_params = {
+            'ldap_id_mapping': 'True',
+            'use_fully_qualified_names': 'False',
+            'debug_level': '9',
+            'id_provider': 'ad',
+            'ad_domain': multihost.ad[0].domainname.lower(),
+            'ad_server': multihost.ad[0].hostname,
+            'ldap_schema': 'ad',
+            'ldap_user_principal': 'userPrincipalName',
+        }
+        client.sssd_conf(dom_section, sssd_params)
+        client.sssd_conf('sssd', {'services': 'nss, pam, ssh'})
+        client.clear_sssd_cache()
+
+        # Create AD user
+        (aduser, _) = create_aduser_group
+
+        # Search for the user
+        usr_cmd = multihost.client[0].run_command(
```

Comment:
This statement 89-92 could be written in single line, now that linewidth is increased.

Reply:
It could, I have editor setup to shorter lines and because of other projects I am not gonna change it.